### PR TITLE
Updated migration script to fix unmatched firm owner data

### PR DESCRIPTION
**Pull request by JazzarKarim**
*Issue #:* /bcgov/entity#19685

*Description of changes:*
- If owner of SP DBA does not match a LEAR business, a colin entity is now created and the DBA is owned by the colin entity
- If a non-person partner entry for a GP does not match a LEAR business, a colin entity is now created and the entity role correctly references the colin entity id.
- Verified changes (below). Please let me know if I'm missing something.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/scripts/manual_db_scripts/legal_name_change/legal_name_updates.sql [hunk 1]
```diff
@@ -403,12 +403,17 @@ select distinct ph.id
...
                     and le.id is null
                     and ph.party_type = 'organization'
                     and (ph.identifier is not null and ph.identifier != '' and ph.identifier not like 'FM%')
-                    )                                                                           is_business_party_colin_entity,
+                    ) or (r.id is null
+                      and pr.filing_id is null
+                      and ph.party_type = 'organization'
+                      and le.id is null
+                      and (ph.identifier is null or ph.identifier = '' or ph.identifier like 'FM%')
+                    ))                                                                          is_business_party_colin_entity,
```
Comment (JazzarKarim) [inline legal-api/scripts/manual_db_scripts/legal_name_change/legal_name_updates.sql:416]: This column `is_business_party_colin_entity` is what dictates the colin entities that'll be created and be linked to.

The idea was to add the condition where when there's no identifier (either '' or null), to drop it in the colin entities table. This is basically what the `is_business_party_org_no_match` dictates. That column though isn't being used anywhere from what I checked.

**Comment by JazzarKarim** [comment]
Some testing/verification:

## Legal Entities table

Legal Entities before, there were 539 entries for type organization:
legalentities before

Legal Entities after, there are 373 entries for type organization now:
legalentities after

As we can see, there were 166 entries that were improperly created as a legal entity not a colin entity. These have been moved to the colin_entities table.

## Colin Entities table

Colin Entities before, there were 407 entries:
colin entities before

Colin Entities after, there are 573 entries now:
colin entities now

Again, the difference is 166 entries. This means that those 166 entries are properly colin entities now.

## Entity Roles table
Now to show proper linking, here's the entity roles table before:
entityroles before
After:
entityroles after

So, proper linking (automatically) is being formed. The script is taking care of that.

One last thing, for partners, here's a small example:
entity_roles 4198 before
entity_roles 4198 after

The last entry is now being properly linked to a colin entity.

**Comment by JazzarKarim** [comment]
[legalentities before organization.csv](https://github.com/bcgov/lear/files/14378116/legalentities.before.organization.csv)
[legalentities after organization.csv](https://github.com/bcgov/lear/files/14378117/legalentities.after.organization.csv)
[colinentities before.csv](https://github.com/bcgov/lear/files/14378121/colinentities.before.csv)
[colinentities after.csv](https://github.com/bcgov/lear/files/14378123/colinentities.after.csv)
[select entityroles relatedcolin not null before.csv](https://github.com/bcgov/lear/files/14378125/select.entityroles.relatedcolin.not.null.before.csv)
[select entityroles relatedcolin not null after.csv](https://github.com/bcgov/lear/files/14378126/select.entityroles.relatedcolin.not.null.after.csv)

(You can neglect these, just a CSV of before and after).

File: legal-api/scripts/manual_db_scripts/legal_name_change/legal_name_updates.sql (hunk 1 above)
Comment (argush3) [reply legal-api/scripts/manual_db_scripts/legal_name_change/legal_name_updates.sql:416]: After thinking about this some more, i think we might be able to simplify this to the following sql.  Anything that doesn't match against LEAR should just go into `colin_entities`.

```sql
(r.id is null
                    and pr.filing_id is null
                    and le.id is null
                    and ph.party_type = 'organization')                                                                           is_business_party_colin_entity
```

Can you update to use this in the two relevant locations in this script and re-run the script to see what numbers we get?

Comment (JazzarKarim) [reply legal-api/scripts/manual_db_scripts/legal_name_change/legal_name_updates.sql]: Argus, I've made the update you mentioned and you're absolutely right. I'm getting the same results.
For legal entities table:
same results legal entities
For colin entities table:
same results colin entities
For entity roles:
entity roles same results

Hence, they're pretty much identical to before.

Comment (argush3) [reply legal-api/scripts/manual_db_scripts/legal_name_change/legal_name_updates.sql]: great, thanks for checking!

**Review by argush3 — APPROVED** [review]

**Pull request merged**